File: cytof/src/scala/cytof/src/main/scala/State.scala
package cytof 

case class State(
  var mus: Array[Array[Double]],    // J x K
  var psi: Array[Double],           // J
  var tau2: Array[Double],          // J
  var pi: Array[Array[Double]],     // I x J
  var c: Array[Double],             // J
  var d: Double, 
  var sig2: Array[Double],          // I
  var v: Array[Double],             // K
  var h: Array[Array[Double]],      // J x K
  var lam: Array[Array[Int]],       // I x N_i
  var w: Array[Array[Double]],      // I x K
  var z: Array[Array[Int]],         // J x K
  var e: Array[Array[Array[Int]]],  // I x N_i x J
  var K: Int) {
}

File: cytof/src/model1/julia/Cytof/src/State.jl
mutable struct State
  mus::Matrix{Float64}          # J x K
  psi::Vector{Float64}          # J
  tau2:: Vector{Float64}        # J
  pi_param::Matrix{Float64}     # I x J
  c::Vector{Float64}            # J
  d::Float64                    # scalar
  sig2::Vector{Float64}         # I
  v::Vector{Float64}            # K
  H::Matrix{Float64}            # J x K
  lam::Array{Array{Int32}}      # I x N_i
  W::Matrix{Float64}            # I x K
  Z::Matrix{Int8}               # J x K
  e_param::Array{Matrix{Int8}}  # I x N_i x J
  K::Int32                      # scalar
end
